pkg/mutation/types/mutator.go

```diff
@@ -1,7 +1,6 @@
-package mutation
+package types
```

Comment:
I was torn about this when I put the structures together.

I like the switch to `schema.Binding`, less having a generic "types" package as the name is not adding information.
mutation.ID / mutation.MakeID sounded way better.
But I don't have a strong opinion either, the only thing I think is, if types is the kitchen sink for all the types, then maybe even SchemaBinding should belong here.

Reply:
I wonder if @shomron has any thoughts on package organization here?

The biggest reason why I did this was so that schema could go into its own package. I felt like between the various tests and the private types that schema needed the `mutation` namespace was becoming a bit cluttered.

Should `types` instead be called something like `common` or similar, to highlight that the only reason it exists is to avoid import cycles?

Agree that mutation.ID sounds more clear. Unfortunately both schema and mutation rely on the ID.

Another thing we could do is make `mutation` a common package and push system, AssignMetatata, etc. into their own sub-packages.

e.g. mutation/mutatotrs/AssignMetadata, mutation/system/system